src/mobile-ui-react/MobileUi.tsx

```diff
@@ -554,15 +613,21 @@ export function useIsolatedCount(): number {
 }
 
 /**
- * A custom React hook function for tracking the UiFramework's current iModel.
+ * A custom React hook function for handling UiFramework iModelConnection changes.
  * @param handler - The callback function.
  */
-export function useIModel(handler: (iModel: IModelConnection | undefined) => void) {
-  useSyncUiEvent(React.useCallback(() => {
-    handler(UiFramework.getIModelConnection());
-    // @todo AppUI deprecation
-    // eslint-disable-next-line deprecation/deprecation
-  }, [handler]), SessionStateActionId.SetIModelConnection);
+export function useIModelChanged(handler: (iModel: IModelConnection | undefined) => void) {
+  useBeEvent(React.useCallback(handler, [handler]), UiFramework.onIModelConnectionChanged);
+}
+
+/**
+ * A custom React hook function for keeping track of the current UiFramework iModelConnection.
+ * @returns The current UiFramework IModelConnection.
+ */
+export function useIModel(): IModelConnection | undefined {
+  const [iModel, setIModel] = React.useState(UiFramework.getIModelConnection());
+  useIModelChanged(setIModel);
```

Comment:
@aruniverse Is it OK to directly pass `setIModel` here without wrapping it in `useCallback`? In other words, is the `setIModel` function that gets returned by `useState` the same for every call to the hook, or does it change every time like a locally declared arrow function would?

Reply:
id hope the setState is memoized by React itself and not something you need to wrap. this should be ok i think